test/dynamo/test_aot_autograd.py

```diff
@@ -940,6 +940,25 @@ def grad_with_create_graph(mod, x, target):
                     bwd_set.add(event.sequence_nr)
         self.assertTrue(len(bwd_set), 13)
 
+    def test_aot_grad_mode_mutation(self):
+        for compiler in ["aot_eager", "inductor"]:
+
+            def f(x):
+                y = x.clone()
+                torch.set_grad_enabled(False)
+                return y.clone()
+
+            f_compiled = torch.compile(f, backend=compiler, fullgraph=True)
+
+            x = torch.randn(3, requires_grad=True)
+            torch.set_grad_enabled(True)
+            y_ref = f(x)
+            self.assertEqual(torch.is_grad_enabled(), False)
+            torch.set_grad_enabled(True)
+            y = f_compiled(x)
+            self.assertEqual(torch.is_grad_enabled(), False)
+            self.assertEqual(y_ref, y)
```

Comment:
You should also test that the grad_fn on the output tensor is what you expected

Reply:
Tested that they are not none and will compute the same tangents for the inputs